EvolvSwiftSDK/EvolvSwiftSDKTests/Common-Tests/EvolvClientTests.swift

```diff
@@ -435,3 +435,120 @@ extension EvolvClientTests {
+        
+        struct Value: Equatable, Hashable {
+            let key: String!
+            let value: String!
+        }
+        
+        var actualValuesAdded = Set<Value>()
+        client.on(topic: CONTEXT_VALUE_ADDED) { value in
+            actualValuesAdded.insert(Value(key: value["key"] as? String, value: value["value"] as? String))
+        }
+        client.on(topic: CONTEXT_VALUE_CHANGED) { value in
+            actualValuesAdded.insert(Value(key: value["key"] as? String, value: value["value"] as? String))
+        }
+        _ = client.set(key: "location", value: "US", local: false)
+        _ = client.set(key: "location", value: "DE", local: true)
+        
+        let expectedValuesAdded: Set<Value> = [.init(key: "location", value: "US"), .init(key: "location", value: "DE")]
+        
+        XCTAssert(expectedValuesAdded.isSubset(of: actualValuesAdded))
+    }
+    
+    func testOnContextValueRemoved() {
+        let options = EvolvClientOptions(evolvDomain: "participants-stg.evolv.ai", participantID: "80658403_1629111253538", environmentId: "4a64e0b2ab", analytics: true, remoteContext: [:], beacon: evolvBeacon)
+        let client = EvolvClientImpl(options: options, evolvAPI: evolvAPI, scope: scope).initialize().wait()
+        
+        struct Value: Equatable, Hashable {
+            let key: String!
+            let value: String?
+        }
+        
+        var actualValuesAdded = Set<Value>()
+        client.on(topic: CONTEXT_VALUE_ADDED) { value in
+            actualValuesAdded.insert(Value(key: value["key"] as? String, value: value["value"] as? String))
+        }
+        client.on(topic: CONTEXT_VALUE_REMOVED) { value in
+            actualValuesAdded.insert(Value(key: value["key"] as? String, value: value["value"] as? String))
+        }
+        _ = client.set(key: "location", value: "US", local: false)
+        _ = client.remove(key: "location")
+        
+        let expectedValuesAdded: Set<Value> = [.init(key: "location", value: "US"), .init(key: "location", value: nil)]
+        
+        XCTAssert(expectedValuesAdded.isSubset(of: actualValuesAdded))
+    }
+    
+    func testOnceContextValueChanged() {
+        let options = EvolvClientOptions(evolvDomain: "participants-stg.evolv.ai", participantID: "80658403_1629111253538", environmentId: "4a64e0b2ab", analytics: true, remoteContext: [:], beacon: evolvBeacon)
+        let client = EvolvClientImpl(options: options, evolvAPI: evolvAPI, scope: scope).initialize().wait()
+        
+        struct Value: Equatable, Hashable {
+            let key: String!
+            let value: String?
+        }
+        
+        var actualValuesAdded = Set<Value>()
+        client.on(topic: CONTEXT_VALUE_CHANGED) { value in
+            actualValuesAdded.insert(Value(key: value["key"] as? String, value: value["value"] as? String))
+        }
+        _ = client.set(key: "location", value: "US", local: false)
```

Comment:
We test that once only fires once -- but not that 'on' fireas repeatedly -- can we test that? My question in an earlier PR was around it looking like 'on' removed itself once firing a single time